### PR TITLE
RStudio IDE and R example

### PR DESCRIPTION
**Pull request by ctmadison**

### CODE REVIEW

File: lang/RStudio/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# RStudio
+
+RStudio is an Integrated Development Environment (IDE) for the R language. By containerizing RStudio, a scientist can leverage the reproducibility and portability of the Singularity platform, allowing them to build data visualizations and analysis tools and ensure that they behave exactly the same in a new environment. 
+
+The definiton file affords some flexibility, allowing you to run RStudio several different ways:
+
+Instance:
+ This will start up rstudio-server, on default port of 8787
```
Comment (jchavezlavalle) [inline lang/RStudio/README.md:8]: Formatting here in markdown:
#### Instance:
This will start up blah blah...

File: lang/RStudio/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+$ singularity run rstudio.sif
+Exec:
+ Any program in the container, but specifically for running R directly.
+
+$ singularity exec rstudio.sif R -f myfile.r
+To build the image run:
+
+$ sudo singularity build rstudio.sif rstudio.def
+When you use run for the IDE, you may need to execute it like:
```
Comment (jchavezlavalle) [inline lang/RStudio/README.md:26]: Of course, again the formatting of markdown for blocks of code so between the triple quote mark.. ```

File: lang/RStudio/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+
+$ SINGULARITYENV_DISPLAY=${DISPLAY} \
+  singularity run rstudio.sif
+This is needed only if the DISPLAY environment variable does not automatically propagate from your host environment to the container.
+
+For a modification of rstudio-server, you can create a rstudio.conf file, and bind mount it into the container. For example to run on port 9999:
+
```
Comment (jchavezlavalle) [inline lang/RStudio/README.md:33]: Reserved words like variables, files, folders and all that stuff can be quoted between quotes for markdown, in this line for example:  `rstudio.conf` between the quote character `

File: lang/RStudio/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+$ echo "www-port=9999" > rserver.conf
+$ singularity instance start \
+  -B rserver.conf:/etc/rstudio/rserver.conf \
+  rstudio.sif rs
+You will then access your rstudio-server instance at:  https://localhost:9999
+
+Related Posts
```
Comment (jchavezlavalle) [inline lang/RStudio/README.md]: Same, check the titles here: 

#### Related Posts

some text here

File: lang/RStudio/RStudio.def [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+From: debian:9
+
+%environment
+    LC_ALL="C"
+    export LC_ALL
+
+%post
+    apt-get update
+    apt-get -y install systemd systemd-sysv gdebi-core procps libssl1.1 ed wget curl libqt5webkit5 libqt5core5a
+    apt-get -y install r-base r-base-dev
```
Comment (jchavezlavalle) [inline lang/RStudio/RStudio.def:11]: These 2 lines (10 and 11) can be merged into one for `apt-get -y install`

File: lang/RStudio/RStudio.def [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    wget -O /rstudio-server-stretch-1.1.456-amd64.deb \
+      https://download2.rstudio.org/rstudio-server-stretch-1.1.456-amd64.deb
+    gdebi -n /rstudio-server-stretch-1.1.456-amd64.deb
+
+    # Setup the "general" CRAN repo
+    echo 'local({
+ r <- getOption("repos")
+ r["CRAN"] = "https://cran.us.r-project.org"
+ options(repos = r)
+})' >> /etc/R/Rprofile.site
```
Comment (jchavezlavalle) [inline lang/RStudio/RStudio.def:25]: Check this indentation here

**Review by jchavezlavalle — COMMENTED** [review]
Markdown formatting and indentation :+1: @ctmadison @jmstover

File: lang/RStudio/RStudio.def (hunk 5 above)
Comment (jmstover) [reply lang/RStudio/RStudio.def:11]: This was done to separate dependencies, from the application itself.

File: lang/RStudio/RStudio.def (hunk 6 above)
Comment (jmstover) [reply lang/RStudio/RStudio.def:25]: It's right when it ends up in the file. ;)

**Review by jmstover — APPROVED** [review]

**Review by jchavezlavalle — APPROVED** [review]
everything looks good! thanks @jmstover @ctmadison

**Pull request merged**